src/ipyradiant/visualization/explore/graph_explorer.py

```diff
@@ -0,0 +1,190 @@
+# Copyright (c) 2021 ipyradiant contributors.
+# Distributed under the terms of the Modified BSD License.
+
+import traitlets as T
+
+import ipywidgets as W
+from rdflib import Graph
+
+from ...basic_tools.custom_uri_ref import CustomURI
+from ...basic_tools.uri_widgets import SelectMultipleURI
+from ...query.api import SPARQLQueryFramer, build_values
+
+
+class AllTypes(SPARQLQueryFramer):
+    sparql = """
+    SELECT DISTINCT ?o
+    WHERE {
+        ?s a ?o .
+    }
+    """
+
+
+class CustomItem:
+    """Class used to build list items with custom repr"""
+
+    def __init__(self, _repr: callable, **kwargs):
+        self._repr = _repr
+        for attr, value in kwargs.items():
+            self.__setattr__(attr, value)
+
+    def __repr__(self):
+        return self._repr(self)
+
+
+class MetaSubjectsOfType(type):
+    """Metaclass to query for type and label for specific VALUES."""
+
+    _sparql = """
+        SELECT DISTINCT ?s ?label
+        WHERE {{
+            ?s a ?type .
+            OPTIONAL {{ ?s rdfs:label ?label . }}
+
+            VALUES ({}) {{
+                {}
+            }}
+        }}
+    """
+    values = None
+
+    @property
+    def sparql(cls):
+        return build_values(cls._sparql, cls.values)
+
+
+class RDFTypeSelectMultiple(W.VBox):
+    """TODO improve docs, and rename attrs and methods?
+    Used to build a selection for available types in the graph.
+    Uses a library query. Uses graph object's namespace.
+    """
+
+    graph = T.Instance(Graph, allow_none=True)
+    label = T.Instance(W.HTML)
+    select_widget = T.Instance(SelectMultipleURI)
+
+    def __init__(self, label: str = None, *args, **kwargs):
```

Comment:
You can use a string as the default:

```def __init__(self, label: str = "Available types:", *args, **kwargs):```

...but it is not good form to hijack the traitlet like this because it confuses the signature...  Is `label` an HTML widget or a string?  Not a big deal for now, but for future reference...

Reply:
Even better would be to keep it consistent with `GraphExploreNodeSelection` and not even have an `__init__`